### ISSUE TITLE
Trailing semicolons before a comment are not detected

### ISSUE THREAD

**Issue by rajivshah3** [realm/SwiftLint#3217]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Trailing semicolons before a comment are not detected by `swiftlint lint`

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.39.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
excluded:
  - Vendor
disabled_rules:
  - opening_brace
  - empty_parentheses_with_trailing_closure
  - function_body_length
  - line_length
  - cyclomatic_complexity
  - large_tuple

# Rule-specific config
trailing_comma:
  mandatory_comma: true
force_try:
  severity: warning
force_cast:
  severity: warning
identifier_name:
  min_length:
    warning: 2
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 11.5, Build version 11E608c
* Do you have a sample that shows the issue? Yes

```swift
// This triggers a violation:
_ = try XMLDocument(data: docData, options: XMLNode.Options()); // Verify that it was generated correctly, which does not always happen!
```
([source](https://github.com/sparkle-project/Sparkle/blob/8a9963bf04789a4165340cbb1eb5baa5df30e23c/generate_appcast/FeedXML.swift#L195))

With comment:
```
$ echo "_ = try XMLDocument(data: docData, options: XMLNode.Options()); // Verify" | swiftlint lint --no-cache --use-stdin --enable-all-rules
Loading configuration from '.swiftlint.yml'
Done linting! Found 0 violations, 0 serious in 1 file.
```

Without comment:
```
$ echo "_ = try XMLDocument(data: docData, options: XMLNode.Options());" | swiftlint lint --no-cache --use-stdin --enable-all-rules 
Loading configuration from '.swiftlint.yml'
<nopath>:1:63: warning: Trailing Semicolon Violation: Lines should not have trailing semicolons. (trailing_semicolon)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by rajivshah3** [comment]
I think this is still an issue

**Comment by mildm8nnered** [comment]
Not sure when this was fixed, but it no longer appears to be a problem as of 0.50.3, so I think this can be closed.

```
% echo "_ = try XMLDocument(data: docData, options: XMLNode.Options()); // Verify" | swiftlint lint --no-cache --use-stdin --enable-all-rules
The `anyobject_protocol` rule is now deprecated and will be completely removed in a future release.
<nopath>:1:63: warning: Trailing Semicolon Violation: Lines should not have trailing semicolons. (trailing_semicolon)
Done linting! Found 1 violation, 0 serious in 1 file.
```
